wa/instruments/perf.py

```diff
@@ -134,5 +146,38 @@ def update_output(self, context):
                                 metric = '{}_{}'.format(label, match.group(3))
                                 context.add_metric(metric, count, classifiers=classifiers)
 
+    def _update_output_record(self, context):
+        outdir = os.path.join(context.output_directory, 'perf')
+        self.collector.get_trace(outdir)
+
+        self.logger.info('Processing perf report reports.')
+        for host_file in os.listdir(outdir):
+            label = host_file.split('.rpt')[0]
+            host_file_path = os.path.join(outdir, host_file)
+            context.add_artifact(label, host_file_path, 'raw')
+            with open(host_file_path) as fh:
+                for line in fh:
+                    words = line.split()
+                    if len(words) >= 1:
+                        if words[0] == '#' and len(words) == 6:
+                            if words[4] == 'event':
+                                event_type = words[5]
+                                event_type = event_type.strip("'")
+                        if words[0] != '#' and '%' in words[0] and len(words) == 5:
+                            metric = 'perf/{}/{}/{}'.format(event_type, words[1], words[2].strip("[]"))
+                            temp = words[0]
+                            count = temp.strip('%')
```

Comment:
`temp` not necessary here. Also, a metric value should be a number, not string.

```
count = numeric(words[0].strip('%'))
```
(note: `numeric` must be imported from `wa.utils.types`.

Reply:
Thanks, changed in latest version. (I initially thought about doing this but expected it to be written out as a string again so didn't seem necessary.)